mlflow/sagemaker/__init__.py

```diff
@@ -1747,6 +1779,480 @@ def _does_model_exist(model_name, sage_client):
+                         status of the pending deployment via native SageMaker APIs or the AWS
+                         console.
+
+                       - timeout_seconds: If ``synchronous`` is ``True``, the deployment process
+                         will return after the specified number of seconds if no definitive result
+                         (success or failure) is achieved. Once the function returns, the caller is
+                         responsible for monitoring the health and status of the pending
+                         deployment using native SageMaker APIs or the AWS console. If
+                         ``synchronous`` is ``False``, this parameter is ignored.
+
+                       - vpc_config: A dictionary specifying the VPC configuration to use when
+                         creating the new SageMaker model associated with this application.
+                         The acceptable values for this parameter are identical to those of the
+                         ``VpcConfig`` parameter in the `SageMaker boto3 client's create_model
+                         method <https://boto3.readthedocs.io/en/latest/reference/services/sagemaker.html
+                         #SageMaker.Client.create_model>`_. For more information, see
+                         https://docs.aws.amazon.com/sagemaker/latest/dg/API_VpcConfig.html.
+
+                         .. code-block:: python
+                             :caption: Example
+
+                             import mlflow.sagemaker as mfs
+                             vpc_config = {
+                                             'SecurityGroupIds': [
+                                                 'sg-123456abc',
+                                             ],
+                                             'Subnets': [
+                                                 'subnet-123456abc',
+                                             ]
+                                          }
+                             client = mfs.SageMakerPlugin("sagemaker")
+                             client.create_deployment(..., config=dict(vpc_config=vpc_config))
+        """
+        final_config = self._default_deployment_config()
+        if config:
+            self._apply_custom_config(final_config, config)
+
+        app_name, flavor = deploy(
+            app_name=name,
+            model_uri=model_uri,
+            flavor=flavor,
+            execution_role_arn=final_config["execution_role_arn"],
+            assume_role_arn=final_config["assume_role_arn"],
+            bucket=final_config["bucket"],
+            image_url=final_config["image_url"],
+            region_name=final_config["region_name"],
+            mode=DEPLOYMENT_MODE_CREATE,
+            archive=final_config["archive"],
+            instance_type=final_config["instance_type"],
+            instance_count=final_config["instance_count"],
```

Comment:
When instance count is passed via the command line, it should still be interpreted as an int, e.g.

```suggestion
            instance_count=int(final_config["instance_count"]),
```

Perhaps it would be better to make this change in `apply_custom_config`.